docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+              "bob": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVlIHVwZGF0ZWQ=",
+        "metadata": {
+          "version": {
+            "block_num": 6
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEYCIQDoFCEmCAirIJHGcIi3/gz6Bd2a+iJdFvzIVRhXYg9RswIhANpSU319PofKI98lydpsjvz5IpIfvUJhKto8ISsrWfeR"
+}
+```
+
+As we can see here, `key1` changed his value three times, in blocks 4, 5, 6
+
+Now lets query `key3` history, using same GET query.
+
+```sh
+bin/signer -data '{"user_id":"alice","db_name":"db2","key":"key3"}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Output**
+```
+MEYCIQDwogJW4wuIBGO5MYVLt1F0nFq+6BsULpKpmCcyGrcx/QIhAMj1uhj6LLW14pF6zZGToP5rdd0tgx9fvwUDHLXLbDO+
+```
+
+Submit query
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEYCIQDwogJW4wuIBGO5MYVLt1F0nFq+6BsULpKpmCcyGrcx/QIhAMj1uhj6LLW14pF6zZGToP5rdd0tgx9fvwUDHLXLbDO+" \
+     -X GET http://127.0.0.1:6001/provenance/data/history/db2/key3 | jq .
+```
+
+**Output**
+
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "values": [
+      {
+        "value": "dGhpcyBpcyBhIHRoaXJkIHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEQCIFE2JIa9MkaGtBEZUqx8MIsYsSSiJS0Atz+TkI6Yl73HAiBRU98GTF+FlQoVXvSuqZArwd5UxjK5aMXBJLEdRCQKTA=="
+}
+```
+
+Key `key3` changed only once, in block 5
+
+> Please note that due `json.Marshal` for protobuf objects limitations, in many cases inside `version` json object, only `block_num` field is presented, without `tx_num`, because `tx_num` equal to zero. But during unmarshalling protobuf object back from json, `tx_num` field filed will be assigned zero value.
+
+### History query with _only_deletes_ option
+As you can see, last query returned value for `<db2, key3>`. Now we want to query if key was deleted at some time, and their last value before delete. We can run same query, but only with `only_deletes` turned `true`
```

Comment:
last query -> the last query
returned value -> returned a value
and their last value -> and its last value
but only with `only_deletes` turned `true` -> but with query parameter `only_deletes` turned `true`

Reply:
What is the list of query parameters legal for this request?

Reply:
>  and its last value

done

Reply:
> last query -> the last query
> returned value -> returned a value
> and their last value -> and its last value
> but only with `only_deletes` turned `true` -> but with query parameter `only_deletes` turned `true`

done